Logic error fails in building cuda_util.cc

**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version: 0.20.0
4. MPI version:
5. CUDA version:  None
6. NCCL version:
7. Python version: 3.8
8. Spark / PySpark version:
9. OS and version: Mageia Linux
10. GCC version: 10.2

**Bug report:**

Compiler fails on non-CUDA builds because of a missing header.  cuda_util.cc does not include <stdexcept>

01-install-python.sh	horovod	  distcc[33436] ERROR: compile /tmp/pip-install-_9_4iuqp/horovod/horovod/mxnet/cuda_util.cc on 172.17.0.1,cpp,lzo failed
01-install-python.sh	horovod	  distcc[33436] (dcc_build_somewhere) Warning: remote compilation of '/tmp/pip-install-_9_4iuqp/horovod/horovod/mxnet/cuda_util.cc' failed, retrying locally
01-install-python.sh	horovod	  distcc[33436] Warning: failed to distribute /tmp/pip-install-_9_4iuqp/horovod/horovod/mxnet/cuda_util.cc to 172.17.0.1,cpp,lzo, running locally instead
01-install-python.sh	horovod	  /tmp/pip-install-_9_4iuqp/horovod/horovod/mxnet/cuda_util.cc: In constructor ‘horovod::mxnet::with_device::with_device(int)’:
01-install-python.sh	horovod	  /tmp/pip-install-_9_4iuqp/horovod/horovod/mxnet/cuda_util.cc:36:16: error: ‘logic_error’ is not a member of ‘std’
01-install-python.sh	horovod	     36 |     throw std::logic_error("Internal error. Requested device context manager "
01-install-python.sh	horovod	        |                ^~~~~~~~~~~
01-install-python.sh	horovod	  distcc[33436] ERROR: compile /tmp/pip-install-_9_4iuqp/horovod/horovod/mxnet/cuda_util.cc on localhost failed
01-install-python.sh	horovod	  gmake[2]: *** [horovod/mxnet/CMakeFiles/mxnet.dir/build.make:472: horovod/mxnet/CMakeFiles/mxnet.dir/cuda_util.cc.o] Error 1
01-install-python.sh	horovod	  gmake[2]: *** Waiting for unfinished jobs....
01-install-python.sh	horovod	  [ 76%] Building CXX object horovod/torch/CMakeFiles/pytor

--- a/horovod/mxnet/cuda_util.cc
+++ b/horovod/mxnet/cuda_util.cc
@@ -13,6 +13,8 @@
 // limitations under the License.
 // =============================================================================
 
+#include <stdexcept>
+
 #if HAVE_CUDA
 #include "cuda_runtime.h"
 #include <mxnet/base.h>

